metabase/metabase
# Update model cache refresh scheduling control

Part of #22538

The current control managing cache refresh schedule appeared to be a bit difficult to grasp (#22839). This PR replaces these controls and lets people pick one of the default schedules (every hour, 2, 3, 6, 12, 24 hours) or set a custom schedule using [quartz-compatible CRON expression](https://www.quartz-scheduler.org/documentation/quartz-2.3.0/tutorials/crontrigger.html) (the only difference is that we don't allow setting the "seconds" and "year" parts)

### To Verify

1. Go to `/admin/settings/caching`
2. Disable model caching if it's enabled
3. Make sure you don't see the "refresh every" control under the model caching toggle
4. Now enable model caching
5. Ensure the "refresh every" control appeared under the toggle
6. Select a few built-in schedules
7. Make sure that for every option, the FE sends `POST /api/persist/set-refresh-schedule { "cron": <cron-expression> }` request
8. Select "Custom"
9. Fill in some random text or invalid CRON expression
10. Make sure you can see an error message, make sure no `POST /api/persist/set-refresh-schedule` were sent
11. Fill in a valid CRON expression, for instance `0 0/6 * * ?`. Make sure the error is gone and `set-refresh-schedule` request was made
12. Refresh the page, make sure the custom schedule is persisted and is displayed correctly

### Demo

https://user-images.githubusercontent.com/17258145/177328991-4e75048c-a301-49ec-b9e2-aca275a204a4.mp4
